draft-packages/menu/KaizenDraft/Menu/components/MenuList.tsx

```diff
@@ -1,25 +1,24 @@
-import React, { useState } from "react"
+import React, { HTMLAttributes, cloneElement, useState } from "react"
 import { v4 } from "uuid"
+import { OverrideClassName } from "@kaizen/component-base"
 import styles from "./MenuList.module.scss"
 
-export type MenuListProps = {
-  heading?: React.ReactNode
+export interface MenuListProps
+  extends OverrideClassName<HTMLAttributes<HTMLUListElement>> {
+  heading?: JSX.Element
   children: React.ReactNode
 }
 
 export const MenuList = (props: MenuListProps): JSX.Element => {
   const [listHeadingID] = useState<string>(v4())
-  const { heading, children } = props
+  const { heading, children, ...restProps } = props
   return (
     <>
-      {heading && (
-        <span className={styles.header} id={listHeadingID}>
-          {heading}
-        </span>
-      )}
+      {heading && cloneElement(heading, { id: listHeadingID })}
```

Comment:
Hm wondering if it's worth checking the element props for a custom id and overriding the internal one?

Just in case someone wants to add their own id and wonders why it's not working.

Reply:
This is the `MenuList` component and we're injecting the ID into the cloned `MenuHeading` here. People can add their own ids to either `MenuList` and that works fine.

Now that you say that, I don't want to allow ids for MenuHeading because it will break the link, I'll adjust the TS there.

Reply:
Wait never mind I can't do that because it needs to accept the id from here.

Reply:
Yeah, so I was thinking if consumers do this:
```
<MenuList heading={<Heading id="pie">Hello</Heading>} />
```

Then internally you can go:
```
heading.props.id ? setListId(heading.props.id) : setListId(v4());
```

Reply:
Ok that took a bit of rejigging because (currently) from the MenuHeading's perspective, it's always going to get an id and it can't determine if it came from the consumer or if it came from the `cloneElement`.

The ID generation was happening on MenuList rather than MenuHeading. In order to move the ID generation to MenuHeading, I need to pass a set state function through to MenuHeading from MenuList.

Have a look at my latest commit and see what you think. I think it does goes against the React ideology that data should only travel down and not up.

Reply:
This still needs to happen in `MenuList` I think:
```
heading.props.id ? setListId(heading.props.id) : setListId(v4());
```

Reply:
Oops why didn't I think of that, that's much easier